### ISSUE TITLE
Support checking on a patch of git diff

### ISSUE THREAD

**Issue by ZSmallX** [diffplug/spotless#1149]
I'd like to use spotless in my project, but need a new start for only new changed code.

Is it possible to check and apply on a patch of git diff output rather than the whole project codes.

I found that Google Java Format can do this.

> To reformat changed lines in a specific patch, use [google-java-format-diff.py](https://github.com/google/google-java-format/blob/master/scripts/google-java-format-diff.py).

See https://github.com/google/google-java-format#from-the-command-line

**Comment by nedtwigg** [comment]
I think our `ratchet` mode is exactly what you need.

- https://github.com/diffplug/spotless/tree/main/plugin-gradle#ratchet
- https://github.com/diffplug/spotless/tree/main/plugin-maven#ratchet

**Comment by ZSmallX** [comment]
Yeah~ Thanks! That's cool and helps a lot! (Sorry that I only searched keyword like: `git diff` and `patch`)

Only more discussion:

I found the `ratchet ` apply works on file-wide.

While Google Java Format does it better, based on git diff and make effect on code snippet.

It means that it will not break the unchanged code unless it has been modified.

It would be better if spotless can support code snippet mode.
